CareKitStore/CareKitStoreTests/OCKStore/TestStore+Sync.swift

```diff
@@ -267,6 +267,40 @@ class TestStoreSync: XCTestCase {
         XCTAssert(localOutcomes == remoteOutcomes)
         XCTAssert(localOutcomes.count == 1)
     }
+    
+    func testTombstoningOutcomePushedToRemote() throws {
+        let testStore = OCKStore(name: "test", type: .inMemory, remote: dummy)
+        dummy.automaticallySynchronizes = false
```

Comment:
So this test should really be for when automaticallySynchronizes = true, but I couldn't figure out how to test this properly.

The problem it's trying to test is:

1) add an outcome to the local store
2) sync that outcome to remote
3) delete the outcome from the local store causing a tombstone
4) the updated tombstoned and the new outcome replacing the previous are synced to the remote

When using `syncAndWait` it seems to synchronize all entities again, in this test case, 3 entities, where there should only be 2

Reply:
I got around the previous issue by adding a new DummyEndpoint (DummyEndpoint2) which increments and merges it’s knowledgeVector which is needed to test the problem of this fix. Running the testcase on previous versions without the fix will fail